### DISCUSSION TITLE
Unable to read from DB with JS SDK

### DISCUSSION THREAD

**Discussion by apetta** [supabase/supabase#4641] [Questions] [answered]
I'm using the js supabase client to interact with the a table that I've made with the supabase UI. I'm able to write to the DB with the SDK, but for some reason I'm unable to read anything I've written. The data is visible in the UI, so a bit confused as to what's going on. I've tried setting and removing RLS/Policy, and also tried both the anon & service_role API keys but no luck. Any guidance is appreciated!

**Comment by steve-chavez** [comment]
> The data is visible in the UI, so a bit confused as to what's going on.

The UI runs under a different role than the SDK. Try disabling RLS.

Related: https://github.com/supabase/supabase/discussions/4643#discussioncomment-1863274

**Comment by apetta** [comment, marked as answer]
Okay so it turns out that "data" is a required variable name in the below snippet (i.e. you can't use another variable name), that was my issue!

`
let { data, error } = await supabase.from('table').select('*')
`

**Reply by GaryAustin1** [reply]
You can use data:notes syntax to assign to a different variable name other than data.

**Reply by apetta** [reply]
> You can use data:notes syntax to assign to a different variable name other than data.

Ahhhh, it's unpacking a single object! That makes sense. Thought it was just returning two separate objects. Thanks for the tip!
